python-package/lightgbm/basic.py

```diff
@@ -848,13 +848,12 @@ def _lazy_init(self, data, label=None, reference=None,
             self.set_weight(weight)
         if group is not None:
             self.set_group(group)
-        # load init score
-        if init_score is not None:
-            self.set_init_score(init_score)
-            if predictor is not None:
-                warnings.warn("The prediction of init_model will be overridden by init_score.")
-        elif isinstance(predictor, _InnerPredictor):
+        if isinstance(predictor, _InnerPredictor):
+            if self._predictor is None and init_score is not None:
```

Comment:
Can you please clarify `if self._predictor is None` check?

Reply:
It means the init_score is set by user, no init_model

Reply:
So, the case where everything `is not None` will silently replace old predictor and the overwrite init_score, right?

Reply:
yeah, replace the init_model will be silent.

Reply:
Got it, thanks!